compass-js-frontend/src/main/webapp/ui/view/editor/hierarchy.js

```diff
@@ -52,10 +52,32 @@ var HierarchyView = AmpersandView.extend({
 		});
 		// what we actually want to save is the internal tree object.
 		this.tree = $tree.fancytree('getTree');
+		$tree.on('contextmenu', this.showContextMenu.bind(this));
+	},
+	registerContextOnMenu: function() {
+		var $menu = $('.basicContextContainer');
+		$menu.on('contextmenu', function() {
+			basicContext.close();
+			return false;
+		});
+	},
+	showContextMenu: function(event, data){
+		var node = $.ui.fancytree.getNode(event);
+		this.selectCurrentNode(node);
+		var items = [
+			{ type: 'item', title: 'Add Node', icon: 'ion-plus-round', fn: this.newChild.bind(this) },
```

Comment:
The selection changes to the node which opened the context menu, so the new node will be created at cursor location (which is the freshly selected scene node ;) )